### PR TITLE
Standalone diarization+ASR evaluation script

### PR DESCRIPTION
**Pull request by tango4j**
# What does this PR do ?

Add a standalone diarization-ASR evaluation transcript.
This script was build at the request for testing ASR+diarization with already extracted files.

Adding diar_infer_general.yaml (VAD optimized for multilingual ASR dataset, diarization optimized on DIHARD dev set)

**Collection**: [Note which collection this PR will affect]

ASR 
# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
python eval_diar_with_asr.py \
 --hyp_rttm_list="/path/to/hypothesis_rttm_filepaths.list" \
 --ref_rttm_list="/path/to/reference_rttm_filepaths.list" \
 --ref_ctm_list="/path/to/reference_ctm_filepaths.list" \
 --hyp_ctm_list="/path/to/hypothesis_ctm_filepaths.list" \
 --root_path="/path/to/output/directory"
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?
Anyone belongs to ASR

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/speaker_tasks/diarization/conf/inference/diar_infer_general.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      onset: 0.5 # Onset threshold for detecting the beginning and end of a speech 
+      offset: 0.3 # Offset threshold for detecting the end of a speech
+      pad_onset: 0.2 # Adding durations before each speech segment 
+      pad_offset: 0.2 # Adding durations after each speech segment 
+      min_duration_on: 0.5 # Threshold for small non_speech deletion
+      min_duration_off: 0.5 # Threshold for short speech segment deletion
+      filter_speech_first: True 
+
+  speaker_embeddings:
+    model_path: titanet_large # .nemo local model path or pretrained model name (titanet_large, ecapa_tdnn or speakerverification_speakernet)
```
Comment (nithinraok) [inline examples/speaker_tasks/diarization/conf/inference/diar_infer_general.yaml:38]: Is it `titanet_large` or `titanet_small` ?

File: nemo/collections/asr/parts/utils/diarization_utils.py [hunk 2]
```diff
@@ -187,6 +187,78 @@ def convert_word_dict_seq_to_ctm(
...
+        total_result_dict[uniq_id]["uniq_id"] = uniq_id
+        if uniq_id in der_results:
+            total_result_dict[uniq_id].update(der_results[uniq_id])
+        if uniq_id in wer_results:
+            total_result_dict[uniq_id].update(wer_results[uniq_id])
+    total_result_jsons = list(total_result_dict.values())
+    return total_result_jsons
+
+
+def get_audacity_label(word: str, stt_sec: float, end_sec: float, speaker: str) -> str:
```
Comment (nithinraok) [inline nemo/collections/asr/parts/utils/diarization_utils.py:223]: Is this moved from speaker_utils? Why?

File: scripts/speaker_tasks/eval_diar_with_asr.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+    for collar, ignore_overlap in eval_settings:
+        diar_score = score_labels(
+            AUDIO_RTTM_MAP=audio_rttm_map_dict,
+            all_reference=all_reference,
+            all_hypothesis=all_hypothesis,
+            collar=collar,
+            ignore_overlap=ignore_overlap,
+        )
+    return diar_score
```
Comment (nithinraok) [inline scripts/speaker_tasks/eval_diar_with_asr.py:151]: shall we keep this part of speaker_utils?

File: examples/speaker_tasks/diarization/conf/inference/diar_infer_general.yaml (hunk 1 above)
Comment (tango4j) [reply examples/speaker_tasks/diarization/conf/inference/diar_infer_general.yaml:38]: If titanet_small is published, we should change this right away

File: nemo/collections/asr/parts/utils/diarization_utils.py (hunk 2 above)
Comment (tango4j) [reply nemo/collections/asr/parts/utils/diarization_utils.py:223]: moved to der.py

File: examples/speaker_tasks/diarization/conf/inference/diar_infer_general.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+batch_size: 64
+
+diarizer:
+  manifest_filepath: ???
+  out_dir: ???
+  oracle_vad: False # If True, uses RTTM files provided in the manifest file to get speech activity (VAD) timestamps
+  collar: 0.25 # Collar value for scoring
+  ignore_overlap: True # Consider or ignore overlap segments while scoring
+
+  vad:
```
Comment (fayejf) [inline examples/speaker_tasks/diarization/conf/inference/diar_infer_general.yaml:20]: This part looks good to me.

File: scripts/speaker_tasks/eval_diar_with_asr.py (hunk 3 above)
Comment (tango4j) [reply scripts/speaker_tasks/eval_diar_with_asr.py]: moved to der.py

File: scripts/speaker_tasks/eval_diar_with_asr.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import json
+import os
+
+from nemo.collections.asr.metrics.der import evaluate_der, score_labels
```
[the remark this answers is not in the document]
Comment (nithinraok) [reply scripts/speaker_tasks/eval_diar_with_asr.py:20]: score_labels is unused here

[the remark this answers is not in the document]
Comment (tango4j) [reply scripts/speaker_tasks/eval_diar_with_asr.py]: fixed

**Review by nithinraok — APPROVED** [review]

**Pull request merged**